### PR TITLE
Add TextEditor

### LINKED ISSUE #47: SwiftUI -> User Interface Element -> Text input and output: TextEditor
Doc: https://developer.apple.com/documentation/swiftui/texteditor

### PR DESCRIPTION
**Pull request by shadowfacts**
Closes #47

### CODE REVIEW

**Review by carson-katri — COMMENTED** [review]
There's a few modifiers that apply to `TextEditor`, not sure if these should be treated as attributes in this case:

* [`findNavigator(isPresented:)`](https://developer.apple.com/documentation/swiftui/view/findnavigator(ispresented:))
* [`findDisabled(_:)`](https://developer.apple.com/documentation/swiftui/view/finddisabled(_:))
* [`replaceDisabled(_:)`](https://developer.apple.com/documentation/swiftui/view/replacedisabled(_:))

**Comment by shadowfacts** [comment]
They should be attributes, however this is blocked by server-client bindings. I intend to focus on that next, so this PR can wait for the moment.

File: Sources/LiveViewNative/BuiltinRegistry.swift [hunk 1]
```diff
@@ -46,6 +46,8 @@ struct BuiltinRegistry {
             Shape(element: element, context: context, shape: RoundedRectangle(from: element))
         case "lvn-link":
             Link(element: element, context: context)
+        case "texteditor":
```
Comment (carson-katri) [inline Sources/LiveViewNative/BuiltinRegistry.swift:49]: If we go with #154 can you rename this to `text-editor`?

```suggestion
        case "text-editor":
```

**Review by carson-katri — CHANGES_REQUESTED** [review]
Some conditionals are missing, so this doesn't build on certain platforms.

File: Sources/LiveViewNative/Views/Text Input and Output/TextEditor.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @FormState var value: String?
+    @FocusState private var isFocused: Bool
+    @LiveBinding(attribute: "find-presented") private var isFindPresented = false
+    
+    init(element: ElementNode, context: LiveContext<R>) {
+        self.context = context
+    }
+    
+    var body: some View {
+        SwiftUI.TextEditor(text: textBinding)
```
Comment (carson-katri) [inline Sources/LiveViewNative/Views/Text Input and Output/TextEditor.swift:22]: FYI, TextEditor is completely unavailable on watchOS

Comment (bcardarella) [reply Sources/LiveViewNative/Views/Text Input and Output/TextEditor.swift:22]: What about for ants?

Comment (shadowfacts) [reply Sources/LiveViewNative/Views/Text Input and Output/TextEditor.swift:22]: 🤦‍♂️ I completely missed that

**Review by carson-katri — APPROVED** [review]

**Pull request merged**